### PR TITLE
add xattr translation

### LINKED ISSUE #70: Problem opening files on OS X
This is a rare issue, rare enough to be difficult to reproduce, but not rare enough to not be annoying. We use version 1.8 installed from homebrew. 

How to reproduce (when this happens):
1. Double-click on the file (in this case PDF, but same happens with Word for example)
2. Dialog window saying "File_1" is damaged and can't be opened. You should eject the disk. It also says "Preview created this file on {date}.", which is correct. There are two options: Cancel and Eject Disk

If I copy this file to /tmp, I can open it, they are even same:

```
$ md5 encfs/File_1.pdf
MD5 (encfs/File_1.pdf) = 0bc06a6fae03f2ca9c4c55cd7cd81ad4
$ md5 /tmp/File_1.pdf
MD5 (/tmp/File_1.pdf) = 0bc06a6fae03f2ca9c4c55cd7cd81ad4
```

The underlying folder is synchronized using Synology CloudStation (private Dropbox alternative), so this may be related. It does not happen with unencrypted folder, so it's either an encfs issue or encfs+Synology CloudStation interaction issue. Internally, it's like dropbox, synchronizes the underlying files when they change (so it's not using FUSE or mount any other filesystem, it's an userspace application). Also noted: This file has been synchronized months ago and has not been changed, so it's probably not a locking issue and for sure not a synchronization conflict issue.

### PR DESCRIPTION
**Pull request by vgough**
Translate extended attributes with special meaning on OSX.

This is based on code from OSXFuse's loopback example.  Intended to fix #70

### CODE REVIEW

**Comment by vgough** [comment]
TODO: com.apple xattrs need to be thrown out when decoding, so that they aren't passed through.  This will effectively strip com.apple xattrs from existing filesystems though.

File: encfs/encfs.cpp [hunk 1]
```diff
@@ -60,18 +62,21 @@ namespace rlog {
...
 using namespace std::placeholders;
 using namespace rlog;
 using rel::Lock;
 
-#define GET_FN(ctx, finfo) ctx->getNode((void *)(uintptr_t) finfo->fh)
+#define GET_FN(ctx, finfo) ctx->getNode((void *)(uintptr_t)finfo->fh)
+
+// XAttr prefix replacement.
+// Replaces "com.apple." with "encfs_apl." when storing xattrs.
+// TODO: future versions of encfs should encrypt xattrs, making this
```
Comment (vgough) [inline encfs/encfs.cpp]: skip this and go straight to xattr encryption?

Comment (benrubson) [reply encfs/encfs.cpp]: Yep.
Did we ever encounter (other) occurrences of the addressed issue ?

Comment (benrubson) [reply encfs/encfs.cpp]: Yes, in addition sounds like #70 is fixed since FUSE for OS X (2.8.2).
So I think we can close this PR yes.

**Pull request closed without merging**